src/write.c

```diff
@@ -1596,6 +1609,9 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
                                               uint32_t gridCols,
                                               uint32_t gridRows,
                                               const avifImage * const * cellImages,
+                                              uint32_t gainMapGridCols,
+                                              uint32_t gainMapGridRows,
+                                              const avifImage * const * gainMapCellImages,
```

Comment:
It would be clearer if we always use the `gainMapCellImages` input parameter when there is a gain map. This requires building the `gainMapCellImages` array in the normal/standard case. If we do this, then a null `gainMapCellImages` means there is no gain map.